sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -396,46 +421,59 @@ public virtual Task<string[]> GetPartitionIdsAsync(CancellationToken cancellatio
                                               SendEventOptions options,
                                               CancellationToken cancellationToken = default)
         {
-            options ??= DefaultSendOptions;
+            bool keepTrying = false;
+            int failedAttemptCount = 0;
 
-            Argument.AssertNotNull(events, nameof(events));
-            AssertSinglePartitionReference(options.PartitionId, options.PartitionKey);
+            do
+            {
+                options ??= DefaultSendOptions;
 
-            // Determine the transport producer to delegate the send operation to.  Because sending to a specific
-            // partition requires a dedicated client, use (or create) that client if a partition was specified.  Otherwise
-            // the default gateway producer can be used to request automatic routing from the Event Hubs service gateway.
+                Argument.AssertNotNull(events, nameof(events));
+                AssertSinglePartitionReference(options.PartitionId, options.PartitionKey);
 
-            TransportProducer activeProducer;
+                // Determine the transport producer to delegate the send operation to.  Because sending to a specific
+                // partition requires a dedicated client, use (or create) that client if a partition was specified.  Otherwise
+                // the default gateway producer can be used to request automatic routing from the Event Hubs service gateway.
 
-            if (string.IsNullOrEmpty(options.PartitionId))
-            {
-                activeProducer = EventHubProducer;
-            }
-            else
-            {
-                // This assertion is intended as an additional check, not as a guarantee.  There still exists a benign
-                // race condition where a transport producer may be created after the client has been closed; in this case
-                // the transport producer will be force-closed with the associated connection or, worst case, will close once
-                // its idle timeout period elapses.
+                TransportProducer activeProducer;
 
-                Argument.AssertNotClosed(IsClosed, nameof(EventHubProducerClient));
-                activeProducer = PartitionProducers.GetOrAdd(options.PartitionId, id => Connection.CreateTransportProducer(id, RetryPolicy));
-            }
+                if (string.IsNullOrEmpty(options.PartitionId))
+                {
+                    activeProducer = EventHubProducer;
+                }
+                else
+                {
+                    // This assertion is intended as an additional check, not as a guarantee.  There still exists a benign
```

Comment:
heh... I was going to mention that I really like this comment.   Then, I realized that I wrote it which is probably why I like it.  😛

Reply:
To tell you the truth, that's something you do frequently.

Reply:
😄